File: Python/Algorithms/Sorting Algorithms/selection_sort.py
numbers = [-2, 45, 0, 11, -9]
unsortedNumbers = [99, 44, 6, 1, 2, 5, 63, 87, 283, 4, 0]


def selection_sort(arr):
    arr_len = len(arr)

    for step in range(arr_len):
        min_idx = step

        for i in range(step + 1, arr_len):
            if arr[i] < arr[min_idx]:
                min_idx = i

        temp = arr[step]
        arr[step] = arr[min_idx]
        arr[min_idx] = temp

        # (arr[step], arr[min_idx]) = (arr[min_idx], arr[step])

    return arr


print(selection_sort(numbers))

File: JavaScript/Algorithms/Sorting/selection_sort.js
const numbers = [-2, 45, 0, 11, -9];
const unsortedNumbers = [99, 44, 6, 1, 2, 5, 63, 87, 283, 4, 0]


function selectionSort(arr) {
  const length = arr.length;

  for (let step = 0; step < length; step++) {
    let min_idx = step;

    for (let i = step + 1; i < length; i++) {
      if (arr[i] < arr[min_idx]) {
        min_idx = i;
      }
    }

    const temp = arr[step];
    arr[step] = arr[min_idx];
    arr[min_idx] = temp;
  }

  return arr;
}

console.log(selectionSort(unsortedNumbers));